### DISCUSSION TITLE
BayesNet Structure Learning

### DISCUSSION THREAD

**Discussion by arainboldt** [ds4dm/ecole#290] [Questions]
Hi all!

Awesome package! I'm exploring the docs and repo and there's a lot to get my head around. I'm particularly curious about the possibility of using the package for structure learning for bayes nets. Unfortunately, I still can't quite understand how exactly I might approach the problem using `ecole` and haven't found any reference to in the docs or discussions. 

Has this been done? Is there any research or examples that could help clarify how this might be done?

I really appreciate any insights and direction.

Thanks!

Andrew

**Comment by AntoinePrv** [comment]
Hi @arainboldt ,

Thank you for your comment! This is a package to solve combinatorics optimization problems ([MILP](https://en.wikipedia.org/wiki/Integer_programming) in particular) using machine learning in combination to the existing solver SCIP.

I'm also not sure how you may use Ecole to do Bayes Net structure learning ! In fact, I'm not so sure how structure learning is done. Can it be written as a combinatorial optimization problem? In such case you could try with a solver, such as SCIP, or using SCIP with Ecole. In the latter case, the learning part in Ecole would still be happening for the optimization process, not directly the for the Bayes net structure.
